### DISCUSSION TITLE
Image in DomPDF

### DISCUSSION THREAD

**Discussion by Nobiscumdeus** [dompdf/dompdf#3043] [Q&A]
Downloaded the DomPdf v 2.0.1 with composer, but there's been issues with the image display. Its not showing locally. Its showing image unknown or not found, how can this be better solved please

**Comment by bsweeney** [comment]
Typically, "image not loading" issues are caused by resource restrictions. For example, if the image is referenced via HTTP protocol (e.g., https://example.com/image.jpg) then the isRemoteEnabled Option has to be set to true and also allow_url_fopen has to be true or the curl extension must be installed (see the [requirements](https://github.com/dompdf/dompdf/wiki/Requirements)). If the image is referenced from the local filesystem (e.g., /var/inet/content/images/image.jpg) then that path must be allowed via the chroot Option (see the [resource validation documentation](https://github.com/dompdf/dompdf/wiki/Usage#resource-references-and-uri-validation)).

So, if the PHP configuration supports the image reference type, then the following Dompdf initialization would configure it to handle resources from both those locations:

```
$dompdf = new Dompdf([
    "isRemoteEnabled" => true,
    "chroot" => ["/var/inet/content"]
]);
```

You can try to get more detail about why things are failing by [enabling additional debug messaging](https://github.com/dompdf/dompdf/wiki/Debugging-Dompdf).

**Reply by Nobiscumdeus** [reply]
Thanks for this ,but how would the image be rightly referenced in the html . I.e <img src=" " />
What would be the content of the src from the example you gave here...

**Reply by Nobiscumdeus** [reply]
What would be the content of the img src in the html

**Reply by Nobiscumdeus** [reply]
Thank you , It worked by having the same content in the Chroot as well as the src of the img tag

**Comment by jaygent** [comment]
Your method doesn't work, can you provide an example code?

**Comment by ashly-dxb** [comment]
I am facing an issue when I include an IMG tag in the HTML view page in a Codeigniter 4 application. The image is not displaying, and instead, a cross icon is appearing in the generated PDF.
I tried with isRemoteEnabled: true, and it is creating an error: Maximum execution time of 60 seconds exceeded .

Any insight will be helpful.

**Reply by bsweeney** [reply]
Did you read through this discussion as well as relevant documentation in the wiki (see [requirements](https://github.com/dompdf/dompdf/wiki/Requirements) and [usage](https://github.com/dompdf/dompdf/wiki/Usage#resource-references-and-uri-validation) documentation)?

Since this discussion is relatively old and your issue possibly unrelated, I recommend starting a new discussion. You'll want to provide as much relevant information as you can such as how you're loading your document, what type of resource reference do you have (local, remote, etc), what type of image, and anything else that might be relevant.
